packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -504,21 +561,53 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
   }
 
   Future<int> _generatePlugin(Directory directory, Map<String, dynamic> templateContext, { bool overwrite = false }) async {
+    // Plugin doesn't create any platform by default
+    templateContext['ios'] = false;
+    templateContext['android'] = false;
+    templateContext['web'] = false;
+    templateContext['linux'] = false;
+    templateContext['macos'] = false;
+    templateContext['windows'] = false;
+    if (argResults.wasParsed('platforms')) {
```

Comment:
if no values are provided, this should give you an empty list, I think the wasParsed check is unecessary

Reply:
Makes sense to me